src/components/vm/hostdevs/hostDevCard.jsx

```diff
@@ -0,0 +1,215 @@
+
+import cockpit from 'cockpit';
+import { vmId, findHostNodeDevice } from "../../../helpers.js";
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+/* Adds an optional description-value pair to an array which represents multiple values of a table cell
+ *
+ * @param chunks an array into which element is added to
+ * @param value a value of the descr-value pair
+ * @param descr a description of the descr-value pair
+ * @param id
+ */
+function addOptionalToCell(chunks, value, id, descr) {
+    if (value) {
+        chunks.push(
+            <div className='machines-multivalue-column' key={descr}>
+                <strong className='machines-multivalue-column-label'>
+                    {descr}
+                </strong>
+                <span className='machines-multivalue-column-value' id={id}>
+                    {value}
+                </span>
+            </div>
+        );
+    }
+}
+
+addOptionalToCell.propTypes = {
+    chunks: PropTypes.array.isRequired,
+    value: PropTypes.oneOf(['string', 'number']),
+    descr: PropTypes.oneOf(['string', 'number']),
+    id: PropTypes.string.isRequired,
+};
+
+export const VmHostDevCard = ({ vm, nodeDevices, config, onAddErrorNotification }) => {
+    function getClass(hostDev, hostdevId) {
+        const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+
+        if (nodeDev && (["usb_device", "pci"].includes(nodeDev.capability.type)))
+            return nodeDev.class;
+    }
+
+    function getProduct(hostDev, hostdevId) {
+        const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+
+        if (nodeDev && (["usb_device", "pci"].includes(nodeDev.capability.type)))
+            return nodeDev.capability.product._value;
+    }
+
+    function getVendor(hostDev, hostdevId) {
+        const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+
+        if (nodeDev && (["usb_device", "pci"].includes(nodeDev.capability.type)))
+            return nodeDev.capability.vendor._value;
+    }
+
+    function getSource(hostDev, hostdevId) {
+        const cell = [];
+        if (hostDev.type === "usb") {
+            const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+            const device = nodeDev.devnum;
+            const bus = nodeDev.busnum;
+
+            addOptionalToCell(cell, device, `${hostdevId}-device`, _("Device"));
+            addOptionalToCell(cell, bus, `${hostdevId}-bus`, _("Bus"));
+        } else if (hostDev.type === "pci") {
+            const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+            const slot = nodeDev.pciSlotName;
+
+            addOptionalToCell(cell, slot, `${hostdevId}-slot`, _("Slot"));
+        } else if (hostDev.type === "scsi") {
+            const bus = hostDev.source.address.bus;
+            const target = hostDev.source.address.target;
+            const unit = hostDev.source.address.lun;
+
+            addOptionalToCell(cell, bus, `${hostdevId}-bus`, _("Bus"));
```

Comment:
I noticed in virt-manager that they reprecent the hos devices like: PCI 0000:06:02.0

domain:\bus:slot:function

Shall we do it also like this to save some space - and not have so many obscure labels?

Reply:
I try to make it closest as possible with what is the result of very popular commands "lspci" and "lsusb". 
- While "lspci" returns a list of pci devices where multiple identifiers are put together into a "slotname" with an annotation like PCI 0000:06:02.0
- the command "lsusb" returns a list of usb devices in a format "Bus 002 Device 002: ID 0bda:0316 Realtek Semiconductor Corp. Card Reader", where it states bus and device explicitely.

Also same for other command, `udevadm`:
- For pci devices it exposes one impotant property: PCI_SLOT_NAME=0000:00:1f.6
- For usb devices however it exposes two properties:
E: BUSNUM=002
E: DEVNUM=002

So I don't feel like trying to invent something new. If it's common to show one property a `slot_name` in a format 0000:00:1f.6 for pci devices, but to show 2 different properties for usb devices, I think we should stick with that